protocol/communities/manager.go

```diff
@@ -4771,6 +4943,91 @@ func (m *Manager) createCommunityTokenPermission(request *requests.CreateCommuni
 	}
 
 	return community, changes, nil
+
+}
+
+func (m *Manager) UpdateControlNode(communityID types.HexBytes, controlNode types.HexBytes) (*Community, error) {
+	community, err := m.GetByID(communityID)
+	if err != nil {
+		return nil, err
+	}
+	if community == nil {
+		return nil, ErrOrgNotFound
+	}
+
+	pubKey, err := crypto.UnmarshalPubkey(controlNode)
+	if err != nil {
+		return nil, err
+	}
+
+	if community.ControlNode().Equal(pubKey) {
+		return community, nil
+	}
+
+	community.setControlNode(pubKey)
+	err = m.persistence.SaveCommunity(community)
+	if err != nil {
+		return nil, err
+	}
+
+	return community, nil
+}
+
+// FIXME: temporary for POC's tests, remove me once communities use timestamp hinted lamport clocks
+func (m *Manager) IncreaseClock(communityID types.HexBytes) (*Community, error) {
+	community, err := m.GetByID(communityID)
+	if err != nil {
+		return nil, err
+	}
+	if community == nil {
+		return nil, ErrOrgNotFound
+	}
+
+	community.increaseClock()
+	community.increaseClock()
+	community.increaseClock()
+
+	err = m.persistence.SaveCommunity(community)
+	if err != nil {
+		return nil, err
+	}
+
+	return community, nil
+}
+
+func (m *Manager) UpdatePrivateKey(communityID types.HexBytes, pk *ecdsa.PrivateKey) (*Community, error) {
+	community, err := m.GetByID(communityID)
+	if err != nil {
+		return nil, err
+	}
+	if community == nil {
+		return nil, ErrOrgNotFound
+	}
+
+	community.setPrivateKey(pk)
+
+	err = m.persistence.SaveCommunity(community)
+	if err != nil {
+		return nil, err
+	}
+
+	return community, nil
+}
+
+// TODO: implement me
```

Comment:
Do you plan to implement this in this PR or should we open a new issue and do it after?

Reply:
oh, it's implemented in a different function, can be removed, good point